samples/Grover/Grover.qs

```diff
@@ -0,0 +1,77 @@
+﻿namespace GroverSample {
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Measurement;
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.Arithmetic;
+
+    open Qram;
+
+    /// # Summary
+    /// This sample is an adaptation of the Grover sample in the QDK documentation
+    /// that uses a BucketBrigadeQRAM as an oracle, rather than the usual reflection
+    /// about marked states.
+    @EntryPoint()
+    operation GroverSearch(addressSize : Int, markedElements : Int[]) : Int {
+        // First, set up a qRAM with marked elements set to 1
+        let nMarkedElements = Length(markedElements);
+        mutable groverMemoryContents = new MemoryCell[0];
+        for (markedElement in markedElements) {
+            set groverMemoryContents += [MemoryCell(markedElement, [true])];
+		}
+        Message($"Num marked: {nMarkedElements}, Num Iterations: {NIterations(nMarkedElements, addressSize)}");
+
+        // Is there a better way to do this in the situation where the memory register
+        // is not explicitly owned by the qRAM?
+        // Also, the target is not really used here except to execute the query and perform
+        // phase kickback
+        using ((groverQubits, targetQubit, flatMemoryRegister) = 
+            (Qubit[addressSize], Qubit[1], Qubit[2^addressSize])
+        ) {
+                // Create a memory
+                // This part feels really clunky
+                let memoryRegister = MemoryRegister(Most(Partitioned(ConstantArray(2^addressSize, 1), flatMemoryRegister)));
+                let memory = BucketBrigadeQRAMOracle(groverMemoryContents, memoryRegister);
+
+                //DumpRegister("memory.txt", flatMemoryRegister);
+                // Initialize a uniform superposition over all possible inputs.
+                PrepareUniform(groverQubits);
+
+                // Grover iterations - the reflection about the marked element is implemented
+                // as a QRAM phase query. Only the memory cells storing a 1 will produce a phase
+                for (idxIteration in 0..NIterations(nMarkedElements, addressSize) - 1) {
+                    //DumpRegister((), flatMemoryRegister);
+                    memory::QueryPhase(AddressRegister(groverQubits), memoryRegister, targetQubit);
+                    //ReflectAboutMarked(groverQubits, markedElement);
+                    ReflectAboutUniform(groverQubits);
+                    //DumpRegister($"reflect_{idxIteration}.txt", groverQubits);
+
+                    // It's necessary to remove phase since QueryPhase only sets phase on the specific address instead of inverting like traditional Grover's
```

Comment:
@glassnotes - What did you think of this fix? Does it seem correct?

Reply:
I think you and @crazy4pi314 made fixes that may no longer require this. I came in mid-stream one day and it seem like you were correcting something.

Reply:
Yes we fixed the issue in stream - it ended up just being a rogue X gate that was supposed to be applied in the context of a within/apply, but instead was sitting outside it.

Re. probabilistic outcome - this is expected. In Grover the space gets split into a superposition of |good/marked stuff> + |bad/not marked> stuff. The rotations boost the amplitude of |good stuff> but this is itself still a superposition so when we measure we'll most likely get one of the good outcomes, and those occur with (I think) equal probability. 

Since it's just a sample I think we can leave it, but in a different application one would indeed have to run it multiple times.

Reply:
Okay, so this code can be removed then. I tested locally and it works correctly.